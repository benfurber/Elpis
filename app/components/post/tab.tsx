import React from "react";
import { StyleSheet, View } from "react-native";

import { TouchableOpacity } from "react-native-gesture-handler";

import { colours } from "styles";
import { Badge, Icon } from "components";

const iconSize = 30;

interface Props {
  name: string;
  number: null | number;
  onPress: () => void;
  selected: boolean;
  first?: boolean;
}

const Tab = (props: Props) => {
  const { first, name, number, onPress, selected } = props;
  const { whiteTransparent, whiteTransparentHigh } = colours;

  const backgroundColor = selected ? whiteTransparent : whiteTransparentHigh;
  const iconColour = selected ? null : colours.mediumGrey;

  const styles = StyleSheet.create({
    background: {
      backgroundColor,
      flex: 1,
    },
    borderLeft: {
      backgroundColor: "transparent",
      borderLeftColor: "transparent",
      borderLeftWidth: 10,
      borderStyle: "solid",
      borderTopColor: backgroundColor,
      borderTopWidth: 46,
    },
    borderRight: {
      backgroundColor: "transparent",
      borderBottomColor: backgroundColor,
      borderBottomWidth: 46,
      borderRightColor: "transparent",
      borderRightWidth: 10,
      borderStyle: "solid",
    },
    iconCentre: {
      alignSelf: "center",
    },
    shiftContainer: {
      flex: 1,
      flexDirection: "row",
      left: first ? 0 : -5,
    },
    touchable: {
      marginVertical: 8,
      paddingHorizontal: 16,
    },
  });

  return (
    <View style={styles.shiftContainer}>
      {!first && <View style={styles.borderLeft} />}
      <View style={styles.background}>
        <TouchableOpacity onPress={onPress}>
          <View style={styles.touchable}>
            <Badge left={65} number={number} />
            <Icon
              colour={iconColour}
              name={name}
              size={iconSize}
              style={styles.iconCentre}
            />
          </View>
        </TouchableOpacity>
      </View>
      <View style={styles.borderRight} />
    </View>
  );
};

export { Tab };
