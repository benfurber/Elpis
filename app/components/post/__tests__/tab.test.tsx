import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { Tab } from "../tab";

describe("Post<Tabs>", () => {
  describe("first tab", () => {
    it("renders correctly when selected", () => {
      const component = shallow(
        <Tab
          name="image"
          onPress={() => jest.fn()}
          selected={true}
          number={null}
          first
        />,
      );
      expect(component).toMatchSnapshot();
    });

    it("renders correctly when not selected", () => {
      const component = shallow(
        <Tab
          name="image"
          onPress={() => jest.fn()}
          selected={false}
          number={null}
          first
        />,
      );
      expect(component).toMatchSnapshot();
    });
  });

  describe("a remaining tab", () => {
    it("renders correctly when selected", () => {
      const component = shallow(
        <Tab
          name="comments"
          onPress={() => jest.fn()}
          selected={true}
          number={23}
        />,
      );
      expect(component).toMatchSnapshot();
    });

    it("renders correctly when not selected", () => {
      const component = shallow(
        <Tab
          name="comments"
          onPress={() => jest.fn()}
          selected={false}
          number={23}
        />,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
