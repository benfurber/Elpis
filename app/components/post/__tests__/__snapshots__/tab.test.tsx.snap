// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post<Tabs> a remaining tab renders correctly when not selected 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "left": -5,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderLeftColor": "transparent",
        "borderLeftWidth": 10,
        "borderStyle": "solid",
        "borderTopColor": "rgba(255, 255, 255, 0.30)",
        "borderTopWidth": 46,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.30)",
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      delayLongPress={600}
      extraButtonProps={
        Object {
          "rippleColor": "transparent",
        }
      }
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "marginVertical": 8,
            "paddingHorizontal": 16,
          }
        }
      >
        <Badge
          left={65}
          number={23}
        />
        <customIcon
          colour="rgba(51,51,51,0.65)"
          name="comments"
          size={30}
          style={
            Object {
              "alignSelf": "center",
            }
          }
        />
      </View>
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "rgba(255, 255, 255, 0.30)",
        "borderBottomWidth": 46,
        "borderRightColor": "transparent",
        "borderRightWidth": 10,
        "borderStyle": "solid",
      }
    }
  />
</View>
`;

exports[`Post<Tabs> a remaining tab renders correctly when selected 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "left": -5,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderLeftColor": "transparent",
        "borderLeftWidth": 10,
        "borderStyle": "solid",
        "borderTopColor": "rgba(255, 255, 255, 0.70)",
        "borderTopWidth": 46,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.70)",
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      delayLongPress={600}
      extraButtonProps={
        Object {
          "rippleColor": "transparent",
        }
      }
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "marginVertical": 8,
            "paddingHorizontal": 16,
          }
        }
      >
        <Badge
          left={65}
          number={23}
        />
        <customIcon
          colour={null}
          name="comments"
          size={30}
          style={
            Object {
              "alignSelf": "center",
            }
          }
        />
      </View>
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "rgba(255, 255, 255, 0.70)",
        "borderBottomWidth": 46,
        "borderRightColor": "transparent",
        "borderRightWidth": 10,
        "borderStyle": "solid",
      }
    }
  />
</View>
`;

exports[`Post<Tabs> first tab renders correctly when not selected 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "left": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.30)",
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      delayLongPress={600}
      extraButtonProps={
        Object {
          "rippleColor": "transparent",
        }
      }
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "marginVertical": 8,
            "paddingHorizontal": 16,
          }
        }
      >
        <Badge
          left={65}
          number={null}
        />
        <customIcon
          colour="rgba(51,51,51,0.65)"
          name="image"
          size={30}
          style={
            Object {
              "alignSelf": "center",
            }
          }
        />
      </View>
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "rgba(255, 255, 255, 0.30)",
        "borderBottomWidth": 46,
        "borderRightColor": "transparent",
        "borderRightWidth": 10,
        "borderStyle": "solid",
      }
    }
  />
</View>
`;

exports[`Post<Tabs> first tab renders correctly when selected 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "left": 0,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.70)",
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      delayLongPress={600}
      extraButtonProps={
        Object {
          "rippleColor": "transparent",
        }
      }
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "marginVertical": 8,
            "paddingHorizontal": 16,
          }
        }
      >
        <Badge
          left={65}
          number={null}
        />
        <customIcon
          colour={null}
          name="image"
          size={30}
          style={
            Object {
              "alignSelf": "center",
            }
          }
        />
      </View>
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "rgba(255, 255, 255, 0.70)",
        "borderBottomWidth": 46,
        "borderRightColor": "transparent",
        "borderRightWidth": 10,
        "borderStyle": "solid",
      }
    }
  />
</View>
`;
